R/derive_var_extreme_flag.R

```diff
@@ -425,74 +295,91 @@ derive_worst_flag <- function(dataset,
 #'   input,
-#'   by_vars = vars(USUBJID, PARAMCD),
-#'   order = vars(ADT),
-#'   new_var = ABLFL,
-#'   mode = "last",
+#'   derivation = derive_var_extreme_flag,
+#'   args = params(
+#'     by_vars = vars(USUBJID, PARAMCD),
+#'     order = vars(ADT),
+#'     new_var = ABLFL,
+#'     mode = "last"),
 #'   filter = AVISIT == "BASELINE"
 #' )
 #'
 #' # Worst observation - Direction = High
-#' derive_var_extreme_flag(
+#' restrict_derivation(
 #'   input,
-#'   by_vars = vars(USUBJID, PARAMCD),
-#'   order = vars(AVAL, ADT),
-#'   new_var = ABLFL,
-#'   mode = "last",
+#'   derivation = derive_var_extreme_flag,
+#'   args = params(
+#'     by_vars = vars(USUBJID, PARAMCD),
+#'     order = vars(AVAL, ADT),
+#'     new_var = ABLFL,
+#'     mode = "last"),
 #'   filter = AVISIT == "BASELINE"
 #' )
 #'
 #' # Worst observation - Direction = Lo
-#' derive_var_extreme_flag(
+#' restrict_derivation(
 #'   input,
-#'   by_vars = vars(USUBJID, PARAMCD),
-#'   order = vars(desc(AVAL), ADT),
-#'   new_var = ABLFL,
-#'   mode = "last",
+#'   derivation = derive_var_extreme_flag,
+#'   args = params(
+#'     by_vars = vars(USUBJID, PARAMCD),
+#'     order = vars(desc(AVAL), ADT),
+#'     new_var = ABLFL,
+#'     mode = "last"),
 #'   filter = AVISIT == "BASELINE"
 #' )
 #'
 #' # Average observation
-#' derive_var_extreme_flag(
+#' restrict_derivation(
 #'   input,
-#'   by_vars = vars(USUBJID, PARAMCD),
-#'   order = vars(ADT, desc(AVAL)),
-#'   new_var = ABLFL,
-#'   mode = "last",
+#'   derivation = derive_var_extreme_flag,
+#'   args = params(
+#'     by_vars = vars(USUBJID, PARAMCD),
+#'     order = vars(ADT, desc(AVAL)),
+#'     new_var = ABLFL,
+#'     mode = "last"),
 #'   filter = AVISIT == "BASELINE" & DTYPE == "AVERAGE"
 #' )
 derive_var_extreme_flag <- function(dataset,
                                 by_vars,
                                 order,
                                 new_var,
                                 mode,
-                                filter = NULL,
+                                filter = deprecated(),
                                 check_type = "warning") {
   new_var <- assert_symbol(enquo(new_var))
   assert_vars(by_vars)
   assert_order_vars(order)
   assert_data_frame(dataset, required_vars = vars(!!!by_vars, !!!extract_vars(order)))
   mode <- assert_character_scalar(mode, values = c("first", "last"), case_sensitive = FALSE)
-  filter <- assert_filter_cond(enquo(filter), optional = TRUE)
   check_type <- assert_character_scalar(
     check_type,
     values = c("none", "warning", "error"),
     case_sensitive = FALSE
   )
-
-  # Select data to consider for flagging
-  if (!quo_is_null(filter)) {
-    data <- dataset %>% filter(!!filter)
-    data_ignore <- dataset %>%
-      filter(!(!!filter) | is.na(!!filter))
-  } else {
-    data <- dataset
+  if (!missing(filter)){
+    warn(paste("`filter` is deprecated as of admiral 0.7.0.",
+               "Please use restict_derivation() instead (see examples).",
```

Comment:
Function name should be in backticks. Also, it should be rest**r**ict.

Reply:
Fixed